pkg/client/unversioned/portforward/portforward.go

```diff
@@ -108,19 +108,20 @@ func parsePorts(ports []string) ([]ForwardedPort, error) {
 }
 
 // New creates a new PortForwarder.
-func New(dialer httpstream.Dialer, ports []string, stopChan <-chan struct{}, out, errOut io.Writer) (*PortForwarder, error) {
+func New(dialer httpstream.Dialer, ports []string, stopChan <-chan struct{}, readyChan chan struct{}, out, errOut io.Writer) (*PortForwarder, error) {
 	if len(ports) == 0 {
 		return nil, errors.New("You must specify at least 1 port")
 	}
 	parsedPorts, err := parsePorts(ports)
 	if err != nil {
 		return nil, err
 	}
+	readyChan = make(chan struct{})
```

Comment:
If the caller is passing in readyChan, I don't think you'd want to throw that away and replace it with a new channel, would you?